service/controller/v3/resource/instance/create.go

```diff
@@ -33,187 +31,205 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	{
-		d, err := deploymentsClient.Get(ctx, key.ClusterID(customObject), vmssDeploymentName)
-		if IsDeploymentNotFound(err) {
-			// fall through
+	if !resourceStatusExists(customObject, "Stage") {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring deployment")
+
+		computedDeployment, err := r.newDeployment(ctx, customObject, nil)
+		if controllercontext.IsInvalidContext(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "missing dispatched output values in controller context")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not ensure deployment")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
```

Comment:
Improved the logging a bit here and there.

Reply:
This feels like a lot of lines, and IMO it is overwhelming when we have to read logs.
Could we somehow reduce this to one line ? Maybe with new key value pairs, to provide more information on a single line.

e.g.
`r.logger.LogCtx(ctx, "level", "debug", "message", "missing dispatched output values in controller context", "action", "canceling resource")`

Reply:
IMO this aligns with https://github.com/giantswarm/fmt/blob/master/go/log_messages.md. I agree it is quite expressive and maybe intense but I like it a lot. These logs make perfect sense when you follow them in production or during development. You know what happens and this is what they are good for.